### ISSUE TITLE
Set application secrets

### ISSUE THREAD

**Issue by ilrock** [fly-apps/terraform-provider-fly#27]
Is it possible at the moment (or will it be possible at some point) to set application-level secrets? It would be cool to have a terraform-based alternative to running

`flyctl secrets set foo=bar -a XXXXXXX`

**Comment by DAlperin** [comment]
Hmm yeah. Hadn't considered but yeah, totally doable. Will add this week.

**Comment by ilrock** [comment]
Amazing stuff. Thank you so much @DAlperin!

**Comment by DAlperin** [comment]
@ilrock out of curiosity how do you currently manage secrets in terraform. Do you use a secrets.tfvar file or some such or do you integrate with a secret service?

**Comment by StarpTech** [comment]
Hi, what's the status? Would love to configure secrets as well.

**Comment by DAlperin** [comment]
@StarpTech coming soon! I've had some health problems, so my work has been limited. But this is definitely coming.

**Comment by StarpTech** [comment]
Thanks for the update. Get well soon!

**Comment by divmgl** [comment]
Any update on this? Just ran into this and the only real alternative is to use some kind of vault and then connect to it using an initial secret.

**Comment by alexnitta** [comment]
I would also love to use this feature. At the moment, I have to do some unsavory workarounds to make sure that secrets exist before I use `terraform apply` to create a machine that relies on them.

**Comment by DAlperin** [comment]
Yeah, I'll take another crack at this later this week. Sorry for the delay. Been crazy buzy.

**Comment by lukas-w** [comment]
I've written a basic implementation of a `fly_secret` resource that may help at https://github.com/lukas-w/terraform-provider-fly/tree/f-secrets.
```terraform
resource "fly_secret" "my_app_secret" {
  app = fly_app.my_app.id
  name = "SECRET_PASSWORD"
  value = "top-secret-value"
}
```

**Comment by lukas-w** [comment]
Turns out that's not a good solution with non-machine apps because it'll trigger a new release & deploy for every single secret. Implemented instead as `secrets` attribute of `fly_app` in PR #106.

**Comment by alexnitta** [comment]
@DAlperin it would be great if you happen to have any time to look at this.
